### ISSUE TITLE
No image returned in options replace: domNode

### ISSUE THREAD

**Issue by linoxyn** [cedricdelpoux/gatsby-source-google-docs#43]
I can properly create pages with the slug and frontmatter though with an `options` image in the Google doc it's not brought into the `<>{htmlContent}</>`

I have in the config optional settings:  `convertImgToNode: true,`
```
const PostTemplate = ({data: {post, googleDocImages}}) => {
    const options = {
        replace: domNode => {
            const {children} = domNode
            if (!children) return // Here I am not returning any children
            const hasImgTag = children.find(img => img.name === "img") // is "img" to be replaced by a variable here? `googleDocImages.edges.node.name ` ? I'm not sure how to handle this.
            if (hasImgTag) {
                const {attribs} = hasImgTag
                const {src, alt} = attribs
                return (
                    <Img
                        fluid={
                            googleDocImages.edges.find(
                                ({node}) => node.id === src
                            ).node.childImageSharp.fluid
                        }
                        alt={alt}
                        className="ui fluid image"
                    />
                )
            }
        },
    }
    const htmlContent = parse(post.html, options)
    return (
        <>
            <h1>{post.frontmatter.name}</h1>
            <p>{post.frontmatter.date}</p>
            <React.Fragment>{htmlContent}</React.Fragment> // if I add a Gatsby `<Img>` tag after this and set the `googleDocImages.edges[2].node.childImageSharp.fluid` the third image loads on the page.
        </>
    )
}
 
```
I can log out the googleDocImages to the console and all the `childImageSharp` data is there within an array of images from the graphql query filtering the `google-doc-image-**` images. 

`googleDocImages.edges.node.name ` gives me the generated gatsby file names though I seem to be going around in circles as to how this snipet of code from the plugin should work. Any hints to the direction to go would be greatly appreciated. Thanks

**Comment by justinsunho** [comment]
is this more of an issue with using https://www.npmjs.com/package/html-react-parser ? seems like all of the stuff that has to do with this plugin works since images are being output when you hardcode it

` if (!children) return // Here I am not returning any children` this line of code is just to stop the replace function if the DOM item doesn't return any children

` const hasImgTag = children.find(img => img.name === "img") // is "img" to be replaced by a variable here? `googleDocImages.edges.node.name ` ? I'm not sure how to handle this.
` this line is to replace a normal html `img` tag with the a gatsby `<Img>` tag


Are the images available locally when `convertImgToNode: false` ?

here is how I'm using it:

```
const Docs = ({data, location}) => {
	const {
		markdownRemark: {
			html,
			frontmatter: {id: googleDocId}
		},
		googleDocImages: {edges: gDocImage}
	} = data;

	const h2Array = [];

	const options = {
		replace: domNode => {
			const {children, name} = domNode;

			if (!children) return;

			//h2 replacement
			if (name === 'h2') {
				const hasText = children.find(h2 => h2.type === 'text');

				if (hasText) {
					const {data: text} = hasText;

					const headerId = camelCase(text);

					h2Array.push({
						id: headerId,
						name: text
					});

					return (
						<ScrollElement id={headerId}>
							<h2
								className='documentation-h2'
								style={{
									marginBottom: '-128px',
									marginTop: '-16px',
									paddingBottom: '136px',
									paddingTop: '16px'
								}}
							>
								{text}
							</h2>
						</ScrollElement>
					);
				}
			}

			//img replacement
			const hasImgTag = children.find(img => img.name === 'img');

			if (hasImgTag) {
				const {attribs} = hasImgTag;
				const {src, alt} = attribs;
				const gDocImageEdge = gDocImage.find(
					({node}) => node.id === src
				);

				if (!!gDocImageEdge) {
					const {
						node: {
							childImageSharp: {fluid}
						}
					} = gDocImageEdge;

					return (
						<Img
							fluid={fluid}
							alt={alt}
							className={`ui fluid image ${styles.image}`}
						/>
					);
				}
			}
		}
	};

	const htmlContent = parse(
		sanitizeHTML(html, {
			allowedTags: sanitizeHTML.defaults.allowedTags.concat([
				'h1',
				'h2',
				'span',
				'img'
			]),
			allowedAttributes: false
		}),
		options
	);
``` 

you can ignore `sanitizeHTML` we're just using that to prevent XSS

**Comment by justinsunho** [comment]
i just took your code and used it for my site and it works, oddly enough. Hopefully that helps

**Comment by linoxyn** [comment]
@justinsunho  Hmmm, interesting. I can't yet seem to trace why I'm not getting this to work. The images in my Google drive that I'm insetting into the docs are public as well as the doc itself. I did check and when `convertImgToNode: false ?` there are no images available locally. I have to have all the config optional settings set to get the images locally. Thank you for your code snippet and I'll try playing with that and get back to you. As you say there may be something else in my code set up that's messing things up. I'll look into `html-react-parser` as well. Thank you :-)

**Comment by linoxyn** [comment]
Thank you @justinsunho It was your hints that the code I posted was fine, and I took a look at any other packages and plugins to rulle them out... Got me thinking.

After digging through as much as I could within my project I had an idea! While I was making my Google Docs, to make things look better laid out I set the text to wrap the image (an image settings feature in Docs). I found that this nullified the node being recognized as an `"img"` Problem solved! Now I can set out to create content, pages, and style to my hearts content now! LOL
